### ISSUE TITLE
`invalid-name` type alias false-positive for union values

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#8487]
### Bug description

With something like:

```python
# pylint: disable=missing-docstring
from typing import Union
x: Union[str, int] = 42
```

pylint claims a wrong type alias name, with no type aliases involved.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
.../test.py:3:0: C0103: Type alias name "x" doesn't conform to predefined naming style (invalid-name)
```

### Expected behavior

No output

### Pylint version

```shell
pylint 2.17.0
astroid 2.15.0
Python 3.10.10 (main, Mar  5 2023, 22:26:53) [GCC 12.2.1 20230201]

also reproducible with the current main branches:

pylint 3.0.0a6
astroid 2.16.0dev0
Python 3.10.10 (main, Mar  5 2023, 22:26:53) [GCC 12.2.1 20230201]
```

**Comment by The-Compiler** [comment]
[tests/functional/r/regression_02/regression_3979.py](https://github.com/PyCQA/pylint/blob/main/tests/functional/r/regression_02/regression_3979.py) with:

```diff
-foo: Union[str, BasePathLike] = "bar"
+Foo: Union[str, BasePathLike] = "bar"
```

**Comment by DanielNoord** [comment]
That fix is because we disallow `foo` as variable name in our configuration and this now gets flagged because we started checking these kind of assignments.

But, this is indeed a false positive that flew under the radar. I'm checking if I can easily fix this!
